qa/tasks/scripts/ses_network_failure.sh

```diff
@@ -0,0 +1,245 @@
+set -ex
+
+pool_name=$(ceph osd pool ls)
+rbd -p $pool_name create image1 --size 2G
+rbd_device=$(rbd map $pool_name/image1)
+parted -s $rbd_device mklabel gpt unit % mkpart 1 xfs 0 100
+mkfs.xfs ${rbd_device}p1
+mount ${rbd_device}p1 /mnt
+
+nettest () {
+set +x
+time_consumed+="$1 - $( (time dd if=/dev/zero  of=/mnt/file1.bin count=100 bs=1M oflag=direct >/dev/null 2>&1) 2>&1 | sed '/^$/d' | head -1); "
+set -x
+ls -l /mnt/file1.bin
+rm -f /mnt/file1.bin
+}
+
+create_netem_rules () {
+
+case $3 in 
+    '') ooo="$(echo -e $2)" ;;
+    *) ooo="$(echo -e "$2" | tail -$3)" ;;
+esac
+
+for minion in $ooo
+do
+    public_interface=$(salt $minion cmd.run "ip -o -4 address | grep $public_octet_ip" | awk '{print $2}' | sed '/^$/d') 
+    cluster_interface=$(salt $minion cmd.run "ip -o -4 address | grep $cluster_octet_ip" | awk '{print $2}' | sed '/^$/d')
+    
+    public_interface_list+=($public_interface)
+    cluster_interface_list+=($cluster_interface)
+    
+    minion_list+=($minion)
+
+    case $1 in 
+        delay|delay_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem delay 100ms 10ms distribution normal"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion  cmd.run "tc qdisc add dev $public_interface root netem delay 100ms 10ms distribution normal"
+            fi
+            ;;
+        packet_loss|packet_loss_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem loss 0.3% 25%"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion cmd.run "tc qdisc add dev $public_interface root netem loss 0.3% 25%"
+            fi
+            ;;
+        packet_dup|packet_dup_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem duplicate 1%"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion cmd.run "tc qdisc add dev $public_interface root netem duplicate 1%"
+            fi
+            ;;
+        packet_corruption|packet_corruption_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem corrupt 0.1%"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion cmd.run "tc qdisc add dev $public_interface root netem corrupt 0.1%"
+            fi
+            ;;
+        packet_reordering|packet_reordering_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem delay 10ms reorder 25% 50%"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion cmd.run "tc qdisc add dev $public_interface root netem delay 10ms reorder 25% 50%"
+            fi
+            ;;
+    esac
+
+done
+}
+
+remove_netem_rules () {
+case $1 in 
```

Comment:
I guess the $1 is some kind of 'mode' or 'operation' or what is this?

Reply:
action to be taken by tc-netem. According this number of minions are used - all or just half.

Reply:
could you please add function key word before this subroutine and add indents to the body, as well as give names for the parameters, so when people read they do not spend much time to understand the code.